+++ b/Makefile
@@ -5,7 +5,7 @@ OCAML_LIBDIR:= $(shell ocamlc -where)
 # GMP_INCLUDES= -I/opt/gmp-4.1.2/include -I/users/absint2/local/include -I$(HOME)/packages/gmp/include

 # GMP_LIBDIR=/opt/gmp-4.1.2/lib
-DESTDIR= $(OCAML_LIBDIR)/gmp
+DESTDIR= $(shell opam var lib)/gmp

 #RLIBFLAGS= -cclib "-Wl,-rpath $(GMP_LIBDIR)" # Linux, FreeBSD
 #RLIBFLAGS= -cclib "-Wl,-R $(GMP_LIBDIR)" # Solaris
@@ -16,7 +16,7 @@ DESTDIR= $(OCAML_LIBDIR)/gmp
 LIBFLAGS = -cclib -L$(shell pwd) -cclib -lgmp -cclib -lmpfr

 #CC= icc
-CFLAGS_MISC= -Wall -Wno-unused -Werror -g -O3
+CFLAGS_MISC= -Wall -Wno-unused -Werror -g -O3 -fcommon
 #CFLAGS_MISC=
 CFLAGS_INCLUDE= -I $(OCAML_LIBDIR) $(GMP_INCLUDES)
 CFLAGS= $(CFLAGS_MISC) $(CFLAGS_INCLUDE)