diff --git a/Makefile b/Makefile
index 2e54a3c..6cee915 100644
--- a/Makefile
+++ b/Makefile
@@ -2,10 +2,10 @@
 RANLIB= ranlib
 
 OCAML_LIBDIR:= $(shell ocamlc -where)
-# GMP_INCLUDES= -I/opt/gmp-4.1.2/include -I/users/absint2/local/include -I$(HOME)/packages/gmp/include
+GMP_INCLUDES= -I$(shell brew --prefix)/include
 
-# GMP_LIBDIR=/opt/gmp-4.1.2/lib
-DESTDIR= $(OCAML_LIBDIR)/gmp
+GMP_LIBDIR= $(shell brew --prefix)/lib
+DESTDIR= $(shell opam var lib)/gmp
 
 #RLIBFLAGS= -cclib "-Wl,-rpath $(GMP_LIBDIR)" # Linux, FreeBSD
 #RLIBFLAGS= -cclib "-Wl,-R $(GMP_LIBDIR)" # Solaris
@@ -13,10 +13,10 @@ DESTDIR= $(OCAML_LIBDIR)/gmp
 
 # LIBFLAGS= -cclib -L. -cclib -L$(GMP_LIBDIR) $(RLIBFLAGS) \
 #	-cclib -lmpfr -cclib -lgmp -cclib -L$(DESTDIR)
-LIBFLAGS = -cclib -L$(shell pwd) -cclib -lgmp -cclib -lmpfr
+LIBFLAGS = -cclib -L. -cclib -lgmp -cclib -lmpfr
 
 #CC= icc
-CFLAGS_MISC= -Wall -Wno-unused -Werror -g -O3
+CFLAGS_MISC= -Wall -Wno-unused -Werror -g -O3 -fcommon
 #CFLAGS_MISC=
 CFLAGS_INCLUDE= -I $(OCAML_LIBDIR) $(GMP_INCLUDES)
 CFLAGS= $(CFLAGS_MISC) $(CFLAGS_INCLUDE)
@@ -86,10 +86,10 @@
 	$(OCAMLOPT) $+ -o $@

 test_suite:	gmp.cma test_suite.cmo
-	$(OCAMLC) -custom $+ -o $@
+	$(OCAMLC) -custom -I $(GMP_LIBDIR) $+ -o $@

 test_suite.opt:	gmp.cmxa test_suite.cmx
-	$(OCAMLOPT) $+ -o $@
+	$(OCAMLOPT) -I $(GMP_LIBDIR) $+ -o $@

 clean:
 	rm -f *.o *.cm* $(PROGRAMS) *.a
